src/python/pants/backend/shell/target_types.py

```diff
@@ -268,12 +269,77 @@ class ShellCommandOutputsField(StringSequenceField):
         Use a trailing slash on directory names, i.e. `my_dir/`.
         """
     )
+    removal_hint = "To fix, use `output_files` and `output_directories` instead."
+    removal_version = "2.17.0.dev0"
+
+
+class ShellCommandOutputFilesField(StringSequenceField):
+    alias = "output_files"
+    required = False
+    default = ()
+    help = softwrap(
+        """
+        Specify the shell command's output files to capture.
+
+        For directories, use `output_directories`. At least one of `output_files` and
+        `output_directories` must be specified.
+        """
+    )
+
+
+class ShellCommandOutputDirectoriesField(StringSequenceField):
+    alias = "output_directories"
+    help = softwrap(
+        """
+        Specify full directories of output from the shell command to capture.
+
+        For files, use `output_files`. At least one of `output_files` and
+        `output_directories` must be specified.
+        """
+    )
 
 
 class ShellCommandDependenciesField(ShellDependenciesField):
     supports_transitive_excludes = True
 
 
+class ShellCommandUseOutputDependenciesInRuntimeField(BoolField):
+    alias = "use_dependencies_at_runtime"
```

Comment:
Maybe the compatibility flag could be specified as a global `use_deprecated_shell_command_dependencies_semantics = ...` flag to flip it globally (similar to https://www.pantsbuild.org/docs/reference-global#use_deprecated_directory_cli_args_semantics).

My understanding of this flag is that it'll require putting `use_dependencies_at_runtime=False` in every single esc when upgrading to the release including this, and then remembering to add it for any new ones (the hardest part 😅 ), and any mistake will lead to noticeably poorer behaviour.